pgml-dashboard/src/components/cards/rgb/mod.rs

```diff
@@ -19,20 +25,44 @@ impl Rgb {
     pub fn new(value: Component) -> Rgb {
         Rgb {
             value,
-            active: false,
             link: None,
+            link_action: CustomOption::default(),
+            controller_classes: vec![],
+            card_classes: vec![],
+            body_classes: vec![],
         }
     }
 
     pub fn active(mut self) -> Self {
-        self.active = true;
+        self.card_classes.push("active".into());
+        self.card_classes.push("main-gradient-border-card-1".into());
+        self
+    }
+
+    pub fn is_active(mut self, active: bool) -> Self {
+        if active {
+            self.card_classes.push("active".into());
+            self.card_classes.push("main-gradient-border-card-1".into());
+        }
+
         self
     }
 
     pub fn link(mut self, link: &str) -> Self {
         self.link = Some(link.to_string());
         self
     }
+
+    pub fn link_action(mut self, action: StimulusAction) -> Self {
+        self.link_action = action.into();
+        self
+    }
+
+    pub fn h_100(mut self) -> Self {
+        self.controller_classes.push("h-100".into());
+        self.card_classes.push("h-100".into());
+        self
```

Comment:
for something like this is it possible to just always have the h-100 on and for instances where you need to limit the height, place it in a wrapper that does not have h-100?  Just brainstorming, this could turn into a slippery slope of having a function for every specific situation.  I think if it is possible to solve problems like this with wrappers, we should.

Reply:
By default, the card occupies as much space as it needs to render its contents. The `.h-100` here helps cards look uniform when placed inside a grid. Our cards don't have the same contents, so their heights will vary when placed inside a grid like `.row` or `.d-flex`. If we have `.h-100` by default, the card will always occupy the full height of its container which is not desired when the card is used by itself inside something like a `<div class="container">`. So `.h-100` is a special case that we just happen to use quite often, but not always.

A wrapper can't make its child occupy 100% of its space unless we add additional CSS to the wrapper knowing that the child needs to do this. For example, here we could do something like:

```sass
.card-wrapper {
    .card {
          @extend .h-100;
    }
}
```

which now that I'm writing this isn't too bad. One issue with this is more of a theoretical technicality, where a component "reaches inside" another component and modifies its state which could cause weird bugs later on. It's typically best to have the component use public methods to modify its state in a predicable  way.

Maybe if we name this slightly differently, it could be better? I'm not leaning strongly one way or the other.